packages/next/build/webpack/plugins/terser-webpack-plugin/src/index.js

```diff
@@ -173,6 +174,11 @@ export class TerserPlugin {
               inputSourceMap = null
             }
 
+            // force dead-code elimination for SPR related code if not used
+            if (!sprStatus.used && file.match(/(runtime|chunks)/)) {
+              input = input.replace('Component.__NEXT_SPR', 'false')
```

Comment:
Replacing strings like this will likely cause an OOM event. Is there a better way we can do this?

Reply:
Yeah, I was hesitant about using `replace` also. I updated it to use the `global_defs` `compress` option of terser. 

Relevant link: https://github.com/terser-js/terser#compress-options